### PR TITLE
Supply mysql8.4.0 by default

### LINKED ISSUE #1051: Supply MySQL 8.2 or 8.3 by default
**Is your feature request related to a problem? Please describe.**
MySQL 8.1 reached [EOL around 6 months ago](https://endoflife.date/mysql)

**Describe the solution you'd like**
Provide MySQL 8.2 or 8.3 by default. Both are still supported.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by rohan-saeed**
fixes #1051
MySQL 8.4.0 LTS has been released. It is set as default in the settings.

Following things has been tested (on dev and k8s) and working perfectly: 
Ability to create user and admin user
Create a course and its visibility on LMS
Ability to login to MySQL with native root user.

### CODE REVIEW

File: tutor/templates/local/docker-compose.yml
Comment (DawoudSheraz) [inline tutor/templates/local/docker-compose.yml:48]: - Reference for this change?
- This would be needed in k8s as well.

**Review by DawoudSheraz — COMMENTED** [review]
Please add some testing instructions. List out some of the things that have been tested.

File: tutor/templates/local/docker-compose.yml
Comment (rohan-saeed) [reply tutor/templates/local/docker-compose.yml:48]: The plugins mysql-native-password is disable by default in mysql 8.4.0 that makes us unable to authenticate with user having native password, such as the root user in our case. See [here](https://dev.mysql.com/doc/refman/8.4/en/native-pluggable-authentication.html) for more context.

Comment (Danyal-Faheem) [reply tutor/templates/local/docker-compose.yml:48]: This option seems to be deprecated and during my testing, I found that removing this did not effect anything in my tutor installation. Can you confirm once again if this is required?

**Review by Danyal-Faheem — CHANGES_REQUESTED** [review]
I've tested these changes on the latest redwood release, and it seems to be working just fine for me. I've even tested my own changes in #1065 with this PR and it seems to be working.

However, I think the `mysql-native-password` option should be removed as it seemed to be working fine for me even after removing it.

File: tutor/templates/local/docker-compose.yml
Comment (rohan-saeed) [reply tutor/templates/local/docker-compose.yml]: Yes, I have tested it again. We are able to authenticate with root user without the plugin.

**Comment by rohan-saeed** [comment]
Changes have been tested on dev environment as well as k8s. The `mysql-native-password` has been removed as it's working fine without it.

**Review by DawoudSheraz — APPROVED** [review]
Played around a bit locally in dev, things seem to be working. 👍🏽

**Review by Danyal-Faheem — APPROVED** [review]
LGTM!

**Comment by regisb** [comment]
@ormsbee did you see this? Would you agree that we upgrade MySQL to 8.4.0 in Redwood, given that 8.1.0 is EOL?

**Comment by ormsbee** [comment]
@regisb: I had not seen it, thank you for tagging me. I agree that it makes sense to go to 8.4 in Redwood.

**Comment by regisb** [comment]
FTR I expect we'll stick with 8.4.0 for as long as possible (2029).

**Comment by DawoudSheraz** [comment]
@regisb Thanks for the discussions on this. Once you give a 👍🏽 , I will merge this.

File: changelog.d/20240531_132815_rohan.saeed_supply_mysql8_4_0.md
Comment (regisb) [inline changelog.d/20240531_132815_rohan.saeed_supply_mysql8_4_0.md:1]: This should be labelled as a breaking change :boom:. Please add a note telling people that the upgrade should be automatic for most people, but if they are running a 3rd-party MySQL (RUN_MYSQL=false) then they are expected to upgrade manually.

**Review by regisb — APPROVED** [review]
This will be good to go once the changelog entry is updated. We should do some hand-holding here :)

File: changelog.d/20240531_132815_rohan.saeed_supply_mysql8_4_0.md
Comment (rohan-saeed) [reply changelog.d/20240531_132815_rohan.saeed_supply_mysql8_4_0.md]: Updated. Please have a look at it again.

**Comment by regisb** [comment]
Looking good :) please make sure to squash all commits during rebase.

**Pull request merged**
